connectors/byoc.py

```diff
@@ -469,6 +469,15 @@ def features(self):
     def last_sync_status(self):
         return JobStatus(self.get("last_sync_status"))
 
+    @property
+    def last_sync_scheduled_at(self):
+        last_sync_scheduled_at = self.get("last_sync_scheduled_at")
+        if last_sync_scheduled_at is not None:
+            last_sync_scheduled_at = datetime.fromisoformat(
+                last_sync_scheduled_at  # pyright: ignore
+            )
+        return last_sync_scheduled_at
```

Comment:
```suggestion
        return datetime.fromisoformat(last_sync_scheduled_at) if last_sync_scheduled_at is not None else last_sync_scheduled_at
```

nit: Just a personal preference, feel free to keep it like that

Reply:
I will keep this for now, we have another property method `last_seen` implemented exactly the same.